app/src/main/java/com/example/una/adapters/StreaksComplexRecyclerViewAdapter.java

```diff
@@ -90,144 +95,42 @@ public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
     }
 
     private void configureChallengeViewHolder(ChallengeViewHolder vhChallenge, int position) {
-        Challenge challenge = (Challenge) items.get(position);
+        Challenge challenge = (Challenge) challenges.get(position);
         if (challenge != null) {
             // set challenge title text view
             vhChallenge.getTvChallengeTitle().setText(challenge.getChallengeName());
             // set challenge owner and associated charity text view
-            setTvOwnerRecipientInfo(vhChallenge, challenge);
-            // set number of participants text view
-            setTvNumParticipants(vhChallenge, challenge);
+            setTvOwnerRecipientInfo(vhChallenge.getTvChallengeOwnerRecipientInfo(), challenge);
+
+            // set participants text view
+            vhChallenge.getTvNumParticipants().setText(getStrNumParticipants(challenge));
+
             // set text view for time left to start or end date of challenge
-            setTvTimeLeft(vhChallenge, challenge);
-            // set challenge progress text view and progress bar
-            setTvProgress(vhChallenge, challenge);
+            vhChallenge.getTvTimeLeft().setText(getStrTimeLeft(challenge));
+
+            // set challenge progress text view
+            vhChallenge.getTvProgress().setText(getStrProgress(challenge));
+
+            // set progress bar
+            setPbProgress(vhChallenge.getPbProgress(), challenge);
 
             // set challenge image placeholder
             int random = (int) (Math.random() * 100 + 1);
+            String url = "https://picsum.photos/id/" + random + "/400/200";
+            challenge.setChallengeImageURl(url);
             Glide.with(context)
```

Comment:
did you make the images permanent?

Reply:
no :/ they only stay the same from the challenges recycler view to challenge details activity (I was thinking we will eventually upload images ourselves to the database, but if not, can make write the URLs to the database!)